lib/Mojolicious/Plugin/OpenAPI.pm

```diff
@@ -257,6 +263,75 @@ sub _route_path {
   return $path;
 }
 
+sub _security_action {
+  my ($self, $api_spec) = @_;
+  my $global      = $api_spec->get('/security') || [];
+  my $definitions = $api_spec->get('/securityDefinitions');
+  my $security_cb = $self->_security_cb;
+
+  return sub {
+    my $c = shift;
+    my @security = @{ $c->openapi->spec->{security} || $global };
+    # return and continue if a requirement set is empty
+    return $c->continue unless @security;
+
+    my ($wrapper, %cache);
+    $wrapper = sub {
+      my $c = shift;
+
+      # return (and don't continue) if no more requirement sets remain
+      return $c->rendered(401) unless @security;
+      my $requirements = shift @security;
+
+      # check cache
+      my @checks;
+      for my $req (keys %$requirements) {
+        my $scopes = $requirements->{$req};
+
+        # build a cache name by joining the security definition name and the scopes with 3 pipes, unlikely in the real world
+        my $name = join '|||', $req, sort @$scopes;
+        my $check = [$name, $security_cb->{$req}, $definitions->{$req}, $scopes];
+
+        if (exists $cache{$name}) {
```

Comment:
The cache test does this already. It defines two different ORed checks each that require `pass1`. The first also requires `fail1` which fails and therefore the second check is attempted. Once that one succeeds and renders, it checks the hash of number of times that it was called, which thus tests that `pass1` was only called once.